code/game/objects/items/devices/laserpointer.dm

```diff
@@ -140,12 +140,19 @@
 	//laser pointer image
 	icon_state = "pointer_[pointer_icon_state]"
 	var/list/showto = list()
-	for(var/mob/M in range(7,targloc))
+	for(var/mob/M in viewers(7,targloc))
 		if(M.client)
 			showto.Add(M.client)
 	var/image/I = image('icons/obj/projectiles.dmi',targloc,pointer_icon_state,10)
-	I.pixel_x = target.pixel_x + rand(-5,5)
-	I.pixel_y = target.pixel_y + rand(-5,5)
+	var/list/click_params = params2list(params)
+	if(click_params)
+		if(click_params["icon-x"])
+			I.pixel_x = text2num(click_params["icon-x"])

```

Comment:
you removed the ```-16```
I should have mentioned you needed to keep them...
sorry, I'm tired.